### PR TITLE
First pass at single-step

### PR DESCRIPTION
**Pull request by gmacd**
- Implement single step (still needs some work)
- Removes some unused code
- Better matches the spec for qSymbol::
- Tries to fix bug when a single register value is requested

### CODE REVIEW

File: sys/src/cmd/gdbserver/debug_core.h [hunk 1]
```diff
@@ -15,16 +15,16 @@
 
 /* kernel debug core data structures */
 struct state {
```
Comment (rminnich) [inline sys/src/cmd/gdbserver/debug_core.h]: when possible it's a good idea to seperate formatting patches from real code.

File: sys/src/cmd/gdbserver/gdbstub.c [hunk 2]
```diff
@@ -918,6 +934,63 @@ gdb_cmd_exception_pass(struct state *ks)
...
+		syslog(0, "gdbserver", "machdata->foll succeeded, nfoll: %d", nfoll);
+
+	for (int i = 0; i < nfoll; i++) {
+		syslog(0, "gdbserver", "Setting single-step breakpoint at: %p", foll[i]);
+		dbg_set_sw_break(ks, foll[i]);
+	}
+
+	gdb_cmd_continue(ks);
+
+	for (int i = 0; i < nfoll; i++) {
```
Comment (rminnich) [inline sys/src/cmd/gdbserver/gdbstub.c]: don't you need to also restore the original one?

Comment (gmacd) [inline sys/src/cmd/gdbserver/gdbstub.c]: Do you mean re-add the breakpoint be just hit?  (The one we explicitly set, not the single step bps)
If so, what happens is that once you continue, gdb single steps, then sends a packet to re-add the breakpoint, before sending another 'c'ontinue packet.  So we don't need to do it ourselved.

File: sys/src/cmd/gdbserver/debug_core.h (hunk 1 above)
Comment (gmacd) [inline sys/src/cmd/gdbserver/debug_core.h]: Yeah, you're right :)

**Pull request merged**